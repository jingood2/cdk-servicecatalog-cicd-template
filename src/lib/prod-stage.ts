import * as cdk from '@aws-cdk/core';
import { envVars } from '../env-vars';
import { SCProductFactoryStack } from './sc-product-factory-stack';

export interface DevStageProps extends cdk.StageProps{

}

export class ProdStage extends cdk.Stage {
  constructor(scope: cdk.Construct, id: string, props: DevStageProps) {
    super(scope, id, props);

    new SCProductFactoryStack(this, `${envVars.SC_PRODUCT_NAME}-stack`);

  }
}