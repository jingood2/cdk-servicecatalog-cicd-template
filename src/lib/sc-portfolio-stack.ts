import * as path from 'path';
import * as iam from '@aws-cdk/aws-iam';
import * as servicecatalog from '@aws-cdk/aws-servicecatalog';
import * as cdk from '@aws-cdk/core';
import { envVars } from '../env-vars';

export enum SCProductType {
  CDK='cdk-sc-product',
  CFN='cfn-sc-product'
}

export class SCProductStack extends cdk.Stack {
  readonly portfolio: servicecatalog.IPortfolio;
  constructor(scope: cdk.Construct, id: string ) {
    super(scope, id );

    if (envVars.SC_PORTFOLIO_ARN != '') {
      this.portfolio = servicecatalog.Portfolio.fromPortfolioArn(this, 'MyImportedPortfolio', envVars.SC_PORTFOLIO_ARN);
    } else {
      this.portfolio = new servicecatalog.Portfolio(this, envVars.SC_PORTFOLIO_NAME, {
        displayName: envVars.SC_PORTFOLIO_NAME ?? 'DemoPortfolio',
        providerName: 'Cloud Infra Team',
        description: 'Portfolio for a project',
        messageLanguage: servicecatalog.MessageLanguage.EN,
      });

      if ( envVars.SC_ACCESS_GROUP_NAME != '') {
        const group = iam.Group.fromGroupName(this, 'SCGroup', envVars.SC_ACCESS_GROUP_NAME);
        this.portfolio.giveAccessToGroup(group);
      }
      if ( envVars.SC_ACCESS_ROLE_ARN != '') {
        const role = iam.Role.fromRoleArn(this, 'SCRole', envVars.SC_ACCESS_ROLE_ARN);
        this.portfolio.giveAccessToRole(role);
      }
    }

    const tagOptionsForPortfolio = new servicecatalog.TagOptions({
      czStage: ['dev', 'qa', 'staging', 'production'],
      czOwner: ['skmagic', 'skens', 'sknetworks', 'skb', 'skbio', 'skcamical'],
    });
    this.portfolio.associateTagOptions(tagOptionsForPortfolio);

    const product = new servicecatalog.CloudFormationProduct(this, 'sc-ec2-linux-ra', {
      productName: 'sc-ec2-linux-ra',
      owner: envVars.SC_PRODUCT_OWNER,
      productVersions: [
        {
          productVersionName: 'v1',
          cloudFormationTemplate: servicecatalog.CloudFormationTemplate.fromAsset(path.join(__dirname, '.', 'cfn-template/ec2/sc-ec2-linux-ra.json')),
        },
      ],
    });

    /* const product = new servicecatalog.CloudFormationProduct(this, 'Product', {
      productName: envVars.SC_PRODUCT_NAME,
      owner: 'Product Owner',
      description: 'test2',
      productVersions: [
        {
          productVersionName: 'v1',
          cloudFormationTemplate: servicecatalog.CloudFormationTemplate.fromProductStack(new ProductConstruct(this, envVars.SC_PRODUCT_NAME)),
        },
      ],
    }); */

    this.portfolio.addProduct(product);
  }
}