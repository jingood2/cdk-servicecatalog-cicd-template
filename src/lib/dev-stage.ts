import * as cdk from '@aws-cdk/core';
import { envVars, SCProductType } from '../env-vars';
import { SCProductStack } from './sc-portfolio-stack';

export interface DevStageProps extends cdk.StageProps{

}

export class DevStage extends cdk.Stage {
  constructor(scope: cdk.Construct, id: string, props: DevStageProps) {
    super(scope, id, props);

    new SCProductStack(this, `${envVars.SC_PRODUCT_NAME}-stack`, {
      portfolioname: 'NewPortfolio',
      codeType: SCProductType.CDK,
    });

  }
}