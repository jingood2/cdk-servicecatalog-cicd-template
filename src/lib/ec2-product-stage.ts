import * as cdk from '@aws-cdk/core';
//import { envVars } from '../env-vars';
import { SCProductStack } from './sc-portfolio-stack';

export interface DevStageProps extends cdk.StageProps{

}

export class EC2ProductStage extends cdk.Stage {
  constructor(scope: cdk.Construct, id: string, props: DevStageProps) {
    super(scope, id, props);

    new SCProductStack(this, 'sc-product', {
      scope: 'ec2',
      env: {
        account: '037729278610',
        region: 'ap-northeast-2', // or whatever region you use
      },
    });
  }
}