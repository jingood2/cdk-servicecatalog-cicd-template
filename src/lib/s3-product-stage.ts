import * as cdk from '@aws-cdk/core';
//import { envVars } from '../env-vars';
import { SCProductStack } from './sc-portfolio-stack';

export interface S3StageProps extends cdk.StageProps{

}

export class S3Stage extends cdk.Stage {
  constructor(scope: cdk.Construct, id: string, props: S3StageProps) {
    super(scope, id, props);

    new SCProductStack(this, 'sc-product', { scope: 's3' } );
  }
}