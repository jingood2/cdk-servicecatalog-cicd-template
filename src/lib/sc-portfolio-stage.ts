import * as cdk from '@aws-cdk/core';
import { SCPortfolioStack } from './sc-portfolio-stack';

export interface ISCPortfolioStageProps extends cdk.StageProps{

}

export class SCPortfolioStage extends cdk.Stage {
  constructor(scope: cdk.Construct, id: string, props: ISCPortfolioStageProps) {
    super(scope, id, props);

    new SCPortfolioStack(this, 'sc-product', {
      env: {
        account: process.env.CDK_DEPLOY_ACCOUNT,
        region: process.env.CDK_DEPLOY_REGION, // or whatever region you use
      },
    });
  }
}