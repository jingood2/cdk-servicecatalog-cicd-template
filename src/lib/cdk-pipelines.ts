import * as codecommit from '@aws-cdk/aws-codecommit';
import * as cdk from '@aws-cdk/core';
import { SecretValue } from '@aws-cdk/core';
import { CodePipeline, CodePipelineSource, ShellStep } from '@aws-cdk/pipelines';
import { envVars } from '../env-vars';
import { SCPortfolioStage } from './sc-portfolio-stage';

export interface CodepipelineSourceProps {
  gitType: string;
  branch: string;
  repoString: string;
  githubToken?: string;
}

export interface CdkPipelinesProps extends cdk.StackProps {
}

export class CdkPipelinesStack extends cdk.Stack {

  constructor(scope: cdk.Construct, id: string, props: CdkPipelinesProps) {
    super(scope, id, props);

    let gitProvider = new cdk.CfnParameter(this, 'git-provider', {
      type: 'String',
      description: 'name of git provider',
      default: 'GITHUB',
      allowedValues: ['GITHUB', 'CODECOMMIT'],
    }).valueAsString;

    gitProvider = envVars.SOURCE_PROVIDER;

    // ðŸ‘‡ parameter of type String
    let repoName = new cdk.CfnParameter(this, 'repository', {
      type: 'String',
      description: 'repository name of cdk infra',
      default: 'jingood2/custom-resource-ddb-example',
    }).valueAsString;

    repoName = envVars.REPO;

    let branchName = new cdk.CfnParameter(this, 'branch', {
      type: 'String',
      description: 'branch name of git repository',
      default: 'master',
    }).valueAsString;

    branchName = envVars.BRANCH;

    new cdk.CfnCondition(this, 'UseGithub', {
      expression: cdk.Fn.conditionEquals(gitProvider, 'github'),
    });

    let githubToken = new cdk.CfnParameter(this, 'githubToken', {
      type: 'String',
      description: 'secret key for github personal access token',
      default: '',
    }).valueAsString;
    githubToken = envVars.GITHUB_TOKEN;

    const pipelinesProps: CodepipelineSourceProps = {
      gitType: gitProvider,
      repoString: repoName,
      branch: branchName,
      githubToken: githubToken,
    };

    const pipeline = new CodePipeline(this, 'Pipeline', {
      selfMutation: true,
      pipelineName: `${envVars.COMPANY_NAME}-${envVars.SC_PRODUCT_NAME}-pipeline`,
      synth: new ShellStep('Synth', {
        input: this.getCodepipelineSource(pipelinesProps),
        commands: [
          'npm i',
          'npm run build',
          'npx cdk synth',
        ],
      }),
    });

    pipeline.addStage(new SCPortfolioStage(this, 'Portfolio', {
      env: {
        account: process.env.CDK_DEPLOY_ACCOUNT,
        region: process.env.CDK_DEPLOY_REGION,
      },
    }));

  }

  private getCodepipelineSource( sourceProps: CodepipelineSourceProps) : CodePipelineSource | undefined {

    switch (sourceProps.gitType) {

      case 'GITHUB':
        return CodePipelineSource.gitHub(sourceProps.repoString, sourceProps.branch, {
          authentication: SecretValue.secretsManager(sourceProps.githubToken ?? ''),
        });
      case 'CODECOMMIT':
        return CodePipelineSource.codeCommit(
          codecommit.Repository.fromRepositoryName(this, 'Repository', sourceProps.repoString), sourceProps.branch);

      default :
        return undefined;

    }
  }
}