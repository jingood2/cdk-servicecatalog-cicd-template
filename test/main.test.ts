import '@aws-cdk/assert/jest';
import { App } from '@aws-cdk/core';
import { SCPortfolioStack } from '../src/lib/sc-portfolio-stack';


test('Snapshot', () => {
  const app = new App();
  const stack = new SCPortfolioStack(app, 'test', {});

  expect(stack).not.toHaveResource('AWS::S3::Bucket');
  expect(app.synth().getStackArtifact(stack.artifactId).template).toMatchSnapshot();
});