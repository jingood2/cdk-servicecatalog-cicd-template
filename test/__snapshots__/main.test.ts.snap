// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters8f3bc81e9754aa133ee6a383cdf0ab792045207af8bcd4478b75a17504e40f2bArtifactHash9243C1B2": Object {
      "Description": "Artifact hash for asset \\"8f3bc81e9754aa133ee6a383cdf0ab792045207af8bcd4478b75a17504e40f2b\\"",
      "Type": "String",
    },
    "AssetParameters8f3bc81e9754aa133ee6a383cdf0ab792045207af8bcd4478b75a17504e40f2bS3BucketFC2A5CA3": Object {
      "Description": "S3 bucket for asset \\"8f3bc81e9754aa133ee6a383cdf0ab792045207af8bcd4478b75a17504e40f2b\\"",
      "Type": "String",
    },
    "AssetParameters8f3bc81e9754aa133ee6a383cdf0ab792045207af8bcd4478b75a17504e40f2bS3VersionKeyFFB3FFD0": Object {
      "Description": "S3 key for asset version \\"8f3bc81e9754aa133ee6a383cdf0ab792045207af8bcd4478b75a17504e40f2b\\"",
      "Type": "String",
    },
    "AssetParametersd9c93ee501f05dbcf2020bde7e6f9cb6e91ef59e608acedc291e6befb547d32aArtifactHash358A7D0A": Object {
      "Description": "Artifact hash for asset \\"d9c93ee501f05dbcf2020bde7e6f9cb6e91ef59e608acedc291e6befb547d32a\\"",
      "Type": "String",
    },
    "AssetParametersd9c93ee501f05dbcf2020bde7e6f9cb6e91ef59e608acedc291e6befb547d32aS3Bucket4D667E3A": Object {
      "Description": "S3 bucket for asset \\"d9c93ee501f05dbcf2020bde7e6f9cb6e91ef59e608acedc291e6befb547d32a\\"",
      "Type": "String",
    },
    "AssetParametersd9c93ee501f05dbcf2020bde7e6f9cb6e91ef59e608acedc291e6befb547d32aS3VersionKey0D3DF610": Object {
      "Description": "S3 key for asset version \\"d9c93ee501f05dbcf2020bde7e6f9cb6e91ef59e608acedc291e6befb547d32a\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "Portfolios3C21F58D3": Object {
      "Properties": Object {
        "AcceptLanguage": "en",
        "Description": "Service Catalog: s3 Reference Architecture",
        "DisplayName": "awstf-portfolios3",
        "ProviderName": "Cloud Infra TF",
      },
      "Type": "AWS::ServiceCatalog::Portfolio",
    },
    "Portfolios3PortfolioProductAssociation3dc39e35cc1c09F8CE7B": Object {
      "Properties": Object {
        "PortfolioId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "ProductId": Object {
          "Ref": "s3bucketandpolicyforcaassev1jsonFE91BE5E",
        },
      },
      "Type": "AWS::ServiceCatalog::PortfolioProductAssociation",
    },
    "Portfolios3PortfolioProductAssociationbf3823887851C980513F": Object {
      "Properties": Object {
        "PortfolioId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "ProductId": Object {
          "Ref": "scs3transitionrajsonCC00FE9C",
        },
      },
      "Type": "AWS::ServiceCatalog::PortfolioProductAssociation",
    },
    "Portfolios3StackSetConstraint3dc39e35cc1cA1741081": Object {
      "DependsOn": Array [
        "Portfolios3PortfolioProductAssociation3dc39e35cc1c09F8CE7B",
      ],
      "Properties": Object {
        "AccountList": Array [
          "856556794427",
        ],
        "AdminRole": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:aws:iam::",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":role/AWSCloudFormationStackSetAdministrationRole",
            ],
          ],
        },
        "Description": "",
        "ExecutionRole": "AWSCloudFormationStackSetExecutionRole",
        "PortfolioId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "ProductId": Object {
          "Ref": "s3bucketandpolicyforcaassev1jsonFE91BE5E",
        },
        "RegionList": Array [
          "ap-northeast-2",
        ],
        "StackInstanceControl": "ALLOWED",
      },
      "Type": "AWS::ServiceCatalog::StackSetConstraint",
    },
    "Portfolios3StackSetConstraintbf382388785106261587": Object {
      "DependsOn": Array [
        "Portfolios3PortfolioProductAssociationbf3823887851C980513F",
      ],
      "Properties": Object {
        "AccountList": Array [
          "856556794427",
        ],
        "AdminRole": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:aws:iam::",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":role/AWSCloudFormationStackSetAdministrationRole",
            ],
          ],
        },
        "Description": "",
        "ExecutionRole": "AWSCloudFormationStackSetExecutionRole",
        "PortfolioId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "ProductId": Object {
          "Ref": "scs3transitionrajsonCC00FE9C",
        },
        "RegionList": Array [
          "ap-northeast-2",
        ],
        "StackInstanceControl": "ALLOWED",
      },
      "Type": "AWS::ServiceCatalog::StackSetConstraint",
    },
    "Portfolios3TagOptionAssociation3eaa5b048498C1AC6AC1": Object {
      "Properties": Object {
        "ResourceId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "TagOptionId": Object {
          "Ref": "TagOption261b16dc50fc",
        },
      },
      "Type": "AWS::ServiceCatalog::TagOptionAssociation",
    },
    "Portfolios3TagOptionAssociation4429e9703575EC1BB3FF": Object {
      "Properties": Object {
        "ResourceId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "TagOptionId": Object {
          "Ref": "TagOption398cbd481f15",
        },
      },
      "Type": "AWS::ServiceCatalog::TagOptionAssociation",
    },
    "Portfolios3TagOptionAssociation44f38caaadbe78A0D531": Object {
      "Properties": Object {
        "ResourceId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "TagOptionId": Object {
          "Ref": "TagOption34c4a21adf82",
        },
      },
      "Type": "AWS::ServiceCatalog::TagOptionAssociation",
    },
    "Portfolios3TagOptionAssociation4832aa8413e5746E0232": Object {
      "Properties": Object {
        "ResourceId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "TagOptionId": Object {
          "Ref": "TagOptionae6b9a23a62c",
        },
      },
      "Type": "AWS::ServiceCatalog::TagOptionAssociation",
    },
    "Portfolios3TagOptionAssociation4fc88847a5808502ED43": Object {
      "Properties": Object {
        "ResourceId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "TagOptionId": Object {
          "Ref": "TagOption672cfa6a2bb5",
        },
      },
      "Type": "AWS::ServiceCatalog::TagOptionAssociation",
    },
    "Portfolios3TagOptionAssociation683878eded9dCB7FE618": Object {
      "Properties": Object {
        "ResourceId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "TagOptionId": Object {
          "Ref": "TagOption6e05492e798e",
        },
      },
      "Type": "AWS::ServiceCatalog::TagOptionAssociation",
    },
    "Portfolios3TagOptionAssociation826063e4626f8AF15CFD": Object {
      "Properties": Object {
        "ResourceId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "TagOptionId": Object {
          "Ref": "TagOption807e7510d8cf",
        },
      },
      "Type": "AWS::ServiceCatalog::TagOptionAssociation",
    },
    "Portfolios3TagOptionAssociation9af08a1fa27a5B78836D": Object {
      "Properties": Object {
        "ResourceId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "TagOptionId": Object {
          "Ref": "TagOption6d2f53e2cb24",
        },
      },
      "Type": "AWS::ServiceCatalog::TagOptionAssociation",
    },
    "Portfolios3TagOptionAssociationc6d43c506f7a0D7707E6": Object {
      "Properties": Object {
        "ResourceId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "TagOptionId": Object {
          "Ref": "TagOption6e6fa7173c5e",
        },
      },
      "Type": "AWS::ServiceCatalog::TagOptionAssociation",
    },
    "Portfolios3TagOptionAssociationfb0f35002140D42E3038": Object {
      "Properties": Object {
        "ResourceId": Object {
          "Ref": "Portfolios3C21F58D3",
        },
        "TagOptionId": Object {
          "Ref": "TagOption71ade984e808",
        },
      },
      "Type": "AWS::ServiceCatalog::TagOptionAssociation",
    },
    "TagOption261b16dc50fc": Object {
      "Properties": Object {
        "Active": true,
        "Key": "czStage",
        "Value": "production",
      },
      "Type": "AWS::ServiceCatalog::TagOption",
    },
    "TagOption34c4a21adf82": Object {
      "Properties": Object {
        "Active": true,
        "Key": "czStage",
        "Value": "qa",
      },
      "Type": "AWS::ServiceCatalog::TagOption",
    },
    "TagOption398cbd481f15": Object {
      "Properties": Object {
        "Active": true,
        "Key": "czOwner",
        "Value": "skbio",
      },
      "Type": "AWS::ServiceCatalog::TagOption",
    },
    "TagOption672cfa6a2bb5": Object {
      "Properties": Object {
        "Active": true,
        "Key": "czOwner",
        "Value": "skb",
      },
      "Type": "AWS::ServiceCatalog::TagOption",
    },
    "TagOption6d2f53e2cb24": Object {
      "Properties": Object {
        "Active": true,
        "Key": "czOwner",
        "Value": "skmagic",
      },
      "Type": "AWS::ServiceCatalog::TagOption",
    },
    "TagOption6e05492e798e": Object {
      "Properties": Object {
        "Active": true,
        "Key": "czStage",
        "Value": "staging",
      },
      "Type": "AWS::ServiceCatalog::TagOption",
    },
    "TagOption6e6fa7173c5e": Object {
      "Properties": Object {
        "Active": true,
        "Key": "czOwner",
        "Value": "sknetworks",
      },
      "Type": "AWS::ServiceCatalog::TagOption",
    },
    "TagOption71ade984e808": Object {
      "Properties": Object {
        "Active": true,
        "Key": "czStage",
        "Value": "dev",
      },
      "Type": "AWS::ServiceCatalog::TagOption",
    },
    "TagOption807e7510d8cf": Object {
      "Properties": Object {
        "Active": true,
        "Key": "czOwner",
        "Value": "skcamical",
      },
      "Type": "AWS::ServiceCatalog::TagOption",
    },
    "TagOptionae6b9a23a62c": Object {
      "Properties": Object {
        "Active": true,
        "Key": "czOwner",
        "Value": "skens",
      },
      "Type": "AWS::ServiceCatalog::TagOption",
    },
    "s3bucketandpolicyforcaassev1jsonFE91BE5E": Object {
      "Properties": Object {
        "Name": "s3-bucket-and-policy-for-caa-sse-v1.json",
        "Owner": "SK CnC AWS TF Team",
        "ProvisioningArtifactParameters": Array [
          Object {
            "DisableTemplateValidation": false,
            "Info": Object {
              "LoadTemplateFromURL": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "https://s3.",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ".",
                    Object {
                      "Ref": "AWS::URLSuffix",
                    },
                    "/",
                    Object {
                      "Ref": "AssetParameters8f3bc81e9754aa133ee6a383cdf0ab792045207af8bcd4478b75a17504e40f2bS3BucketFC2A5CA3",
                    },
                    "/",
                    Object {
                      "Fn::Select": Array [
                        0,
                        Object {
                          "Fn::Split": Array [
                            "||",
                            Object {
                              "Ref": "AssetParameters8f3bc81e9754aa133ee6a383cdf0ab792045207af8bcd4478b75a17504e40f2bS3VersionKeyFFB3FFD0",
                            },
                          ],
                        },
                      ],
                    },
                    Object {
                      "Fn::Select": Array [
                        1,
                        Object {
                          "Fn::Split": Array [
                            "||",
                            Object {
                              "Ref": "AssetParameters8f3bc81e9754aa133ee6a383cdf0ab792045207af8bcd4478b75a17504e40f2bS3VersionKeyFFB3FFD0",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                ],
              },
            },
            "Name": "v1",
          },
        ],
      },
      "Type": "AWS::ServiceCatalog::CloudFormationProduct",
    },
    "scs3transitionrajsonCC00FE9C": Object {
      "Properties": Object {
        "Name": "sc-s3-transition-ra.json",
        "Owner": "SK CnC AWS TF Team",
        "ProvisioningArtifactParameters": Array [
          Object {
            "DisableTemplateValidation": false,
            "Info": Object {
              "LoadTemplateFromURL": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "https://s3.",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ".",
                    Object {
                      "Ref": "AWS::URLSuffix",
                    },
                    "/",
                    Object {
                      "Ref": "AssetParametersd9c93ee501f05dbcf2020bde7e6f9cb6e91ef59e608acedc291e6befb547d32aS3Bucket4D667E3A",
                    },
                    "/",
                    Object {
                      "Fn::Select": Array [
                        0,
                        Object {
                          "Fn::Split": Array [
                            "||",
                            Object {
                              "Ref": "AssetParametersd9c93ee501f05dbcf2020bde7e6f9cb6e91ef59e608acedc291e6befb547d32aS3VersionKey0D3DF610",
                            },
                          ],
                        },
                      ],
                    },
                    Object {
                      "Fn::Select": Array [
                        1,
                        Object {
                          "Fn::Split": Array [
                            "||",
                            Object {
                              "Ref": "AssetParametersd9c93ee501f05dbcf2020bde7e6f9cb6e91ef59e608acedc291e6befb547d32aS3VersionKey0D3DF610",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                ],
              },
            },
            "Name": "v1",
          },
        ],
      },
      "Type": "AWS::ServiceCatalog::CloudFormationProduct",
    },
  },
}
`;
